plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -111,6 +114,57 @@ public function setTitle($title)
     }
 
     /**
+     * @param string $key
+     * @param string $value
+     * @param bool $override
+     *
+     * @return bool
+     */
+    public function addQueryParameter($key, $value, $override = true)
+    {
+        if (!isset($this->queryParameters[$key]) || true === $override) {
+            $this->queryParameters[$key] = $value;
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param array $items
+     * @param bool|true $override
+     */
+    public function addQueryParameters(array $items, $override = true)
+    {
+        foreach ($items as $key => $value) {
+            $this->addQueryParameter($key, $value, $override);
+        }
+    }
+
+    /**
+     * @param $key
+     */
+    public function removeQueryParameter($key)
+    {
+        unset($this->queryParameters[$key]);
+    }
+
+    /**
+     * @param bool $urlEncode
+     * @return array
+     */
+    public function getQueryParameters($urlEncode = false)
```

Comment:
Is the optional parameter needed? Here it's always `$urlEncode=true` and think it would be good if it was always URL encoded just to be safe. Also does it have to be public?